raizama/core/tests/test_command_import_spreadsheet.py

```diff
@@ -0,0 +1,219 @@
+from io import StringIO
+from pathlib import Path
+from django.core.management import call_command
+from django.test import TestCase
+from raizama.core.models import Organization
+from raizama.settings import BASE_DIR
+
+
+class ImportSpreadSheetTest(TestCase):
+    def setUp(self):
+        self.csv_path = Path(
+            BASE_DIR,
+            "raizama",
+            "core",
+            "tests",
+            "fixtures",
+            "OrganizacaoEndEContatos-CasosDeTeste.csv",
+        )
+        self.non_csv_file = "WrongFilePath.txt"
+
+    def test_existance_custom_command(self):
+        self.assertIsNone(call_command("importspreadsheet", self.csv_path))
+
+    def test_exception_non_csv_file(self):
+        self.assertRaises(
+            FileNotFoundError, call_command, "importspreadsheet", self.non_csv_file
+        )
+
+    def test_number_of_organizations(self):
+        call_command("importspreadsheet", self.csv_path)
+        expected_number_orgs = 6
+        output_orgs = Organization.objects.count()
+        self.assertEqual(
+            output_orgs,
+            expected_number_orgs,
+            msg=f"Expected {expected_number_orgs}, but got this {output_orgs}",
+        )
+
+    def test_name_of_first_organization(self):
+        call_command("importspreadsheet", self.csv_path)
+        expected_name = "Uma Doce Revolução"
+        output_name = Organization.objects.first().name
+        self.assertEqual(
+            output_name,
+            expected_name,
+            msg=f"Expected {expected_name}, but got this {output_name}",
+        )
+
+    def test_name_of_last_organization(self):
+        call_command("importspreadsheet", self.csv_path)
+        expected_city = "Copirecê"
+        outpu_city = Organization.objects.last().name
+        self.assertEqual(
+            outpu_city,
+            expected_city,
+            msg=f"Expected {expected_city}, but got this {outpu_city}",
+        )
+
+    def test_city_of_last_oragnization(self):
+        call_command("importspreadsheet", self.csv_path)
+        expected_city = "Irecê"
+        output_city = Organization.objects.last().city
+        self.assertEqual(
+            output_city,
+            expected_city,
+            msg=f"Expected {expected_city}, but got this {output_city}",
+        )
+
+    def test_filter_duplicate(self):
+        call_command("importspreadsheet", self.csv_path)
+        out = StringIO()
+        call_command("importspreadsheet", self.csv_path, stdout=out)
+        self.assertIn("Added 0 new organization(s)", out.getvalue())
```

Comment:
É boa prática usar um gerenciador de contexto para sempre fechar um objeto IO após o uso:

```suggestion
        with StringIO() as out:
            call_command("importspreadsheet", self.csv_path, stdout=out)
            self.assertIn("Added 0 new organization(s)", out.getvalue())
```